dashboards-observability/public/components/event_analytics/hooks/use_fetch_patterns.ts

```diff
@@ -46,47 +51,66 @@ export const useFetchPatterns = ({ pplService, requestParams }: IFetchPatternsPa
+  const buildPatternDataQuery = (query: string, patternField: string) => {
+    return (
+      `${query.trim()} | patterns \`${patternField}\` | ` +
+      `stats count(), take(\`${patternField}\`, 1) by patterns_field`
+    );
+  };
+
+  const buildPatternAnomaliesQuery = (
+    query: string,
+    patternField: string,
+    timestampField: string,
+    interval: string
+  ) => {
+    return (
+      `${query.trim()} | patterns \`${patternField}\` | ` +
+      `stats count() by span(\`${timestampField}\`, 1${interval || 'm'}) as timestamp, ` +
+      `patterns_field | ` +
+      `AD time_field='timestamp' category_field='patterns_field'`
+    );
   };
 
-  const getPatterns = (errorHandler: (error: any) => void, query?: string) => {
+  const getPatterns = (interval: string, errorHandler: (error: any) => void, query?: string) => {
     const cur = queriesRef.current;
     const rawQuery = cur![requestParams.tabId][FINAL_QUERY];
     const searchQuery = isUndefined(query) ? rawQuery : query;
     const patternField = cur![requestParams.tabId][SELECTED_PATTERN];
+    const timestampField = cur![requestParams.tabId][SELECTED_TIMESTAMP];
     const statsQuery = buildPatternDataQuery(searchQuery, patternField);
+    const anomaliesQuery = buildPatternAnomaliesQuery(
+      searchQuery,
+      patternField,
+      timestampField,
+      interval
+    );
     // Fetch patterns data for the current query results
-    fetchEvents(
-      { query: statsQuery },
-      'jdbc',
-      (res: { datarows: any[] }) => {
-        if (!isEmpty(res.datarows)) {
-          const formatToTableData = res.datarows.map((row: any) => {
-            return {
-              count: row[0],
-              pattern: row[2],
-              sampleLog: row[1][0],
-            } as PatternTableData;
-          });
-          // Fetch total number of events to divide count by for ratio
-          fetchEvents(
-            {
-              query: `${rawQuery} | stats count()`,
-            },
-            'jdbc',
-            (countRes: any) => {
-              dispatchOnPatterns({
-                patternTableData: formatToTableData,
-                total: countRes.datarows[0],
-              });
-            },
-            errorHandler
-          );
+    Promise.all([
+      fetchEvents({ query: statsQuery }, 'jdbc', (res) => res, errorHandler),
+      fetchEvents({ query: anomaliesQuery }, 'jdbc', (res) => res, errorHandler),
+      fetchEvents({ query: `${rawQuery} | stats count()` }, 'jdbc', (res) => res, errorHandler),
+    ]).then((res) => {
+      const [statsRes, anomaliesRes, countRes] = res as IPPLEventsDataSource[];
+      const anomaliesMap: { [x: string]: number } = {};
+      anomaliesRes.datarows.forEach((row) => {
+        const pattern = row[2];
+        const score = row[3];
+        if (score > 0) {
+          anomaliesMap[pattern] = (anomaliesMap[pattern] || 0) + 1;
```

Comment:
What's up with + 1 here?

Reply:
Is anomalies map will not be aggregated - each anomaly will show up separately and then added up at the end?

Reply:
not sure what you mean? if anomaly score is 0 we increment the anomaly count for that pattern

Reply:
AD command returns a score per time interval per pattern